### DISCUSSION TITLE
JSON for jsoup

### DISCUSSION THREAD

**Discussion by magicprinc** [jhy/jsoup#2076] [Show and tell]
Just a humble reminder 😅🙏
You can easily parse and query absolutely any JSON (text) with [json4jsoup](https://github.com/magicprinc/jsoup.json)

I've just tested with JSoup 1.17.1 and added `jitpack.io`:
`implementation 'com.github.magicprinc:jsoup.json:1.17.1'`

**Comment by jhy** [comment]
Looks cool, thanks for the reminder!

BTW as feedback, you might like to add an explicit LICENSE file to the project. It took a little diving to find it's Apache 2.0 I believe.

**Reply by magicprinc** [reply]
I need some help in this area:

I personally have no preferences and could add any license.

But I started the project after long meditation on https://github.com/google/gson
and they use Apache 🤔

Can/should I change?

**Reply by jhy** [reply]
Yes I think if the library contains code that was ported from gson you will need to maintain their license and initial copyright.

So the header that you've got in [/magicprinc/jsoup.json/parser/JsonTreeBuilder.java](https://github.com/magicprinc/jsoup.json/blob/master/src/main/java/org/jsoup/parser/JsonTreeBuilder.java) will probably serve as the license.

If the code wasn't a port or you weren't distributing it directly in the repo (doesn't make sense in this library but perhaps for others), personally I would suggest either Apache 2.0 anyway, or MIT as it's very simple and open, while still maintaining the author's credit & copyright.

(Obviously these are the opinions of a non-lawyer, take it with a grain of salt.)

**Reply by magicprinc** [reply]
Done! Apache
